### PR TITLE
MARKET-1300 Copy Button

### PR DESCRIPTION
**Pull request by StasivPavlo**
copy
copied

### CODE REVIEW

**Review by addKiwi — COMMENTED** [review]
remove thumbnail.png

File: components/bl-copyButton-component/component.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      "handlerId": "contentLogic",
+      "handlerLabel": "Content Logic",
+      "dataBinding": true,
+      "type": "text",
+      "handlerDescription": "This is the handler for the logic of determining the content to be copied."
+    },
+    {
+      "type": "text",
+      "name": "copyLabel",
+      "label": "Copy Label",
```
Comment (v-excelsior) [inline components/bl-copyButton-component/component.json:31]: Could we also add Copy Icon with Icon property type? Same for copied icon

File: components/bl-copyButton-component/src/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function CopyButton({ component, eventHandlers }) {
+  const { display, classList, isDisabled, content, copyLabel, copiedLabel, copiedDuration } = component;
+  const { onCopy } = eventHandlers;
+
+  const [isCopied, setIsCopied] = useState(false);
+
+  component.copy = () => {
+    navigator.clipboard.writeText(content).then(() => {
```
Comment (v-excelsior) [inline components/bl-copyButton-component/src/index.js:12]: Do this solution works well on every popular mobile and desktop browsers?

File: components/bl-copyButton-component/src/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      }, copiedDuration);
+    });
+  };
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div className={ cn('bl-customComponent-copyButton', classList) }>
```
Comment (v-excelsior) [inline components/bl-copyButton-component/src/index.js:26]: We need this wrapper? That may cause some problems when Vladimir added default stylable properties

File: components/bl-copyButton-component/component.json (hunk 1 above)
Comment (v-excelsior) [reply components/bl-copyButton-component/component.json:31]: Or we could make different types: `default` | `icon` | `complex`(text and icon in same time)

File: components/bl-copyButton-component/src/index.js (hunk 2 above)
Comment (StasivPavlo) [reply components/bl-copyButton-component/src/index.js:22]: I think that yes
image

File: components/bl-copyButton-component/component.json [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      "handlerId": "copiedLabelLogic",
+      "handlerLabel": "Copied Label Logic",
+      "dataBinding": true,
+      "handlerDescription": "This is the handler for the logic to determine the label for copied button.",
+      "showInSettings": true,
+      "hasLogicHandler": true
+    },
+    {
+      "type": "checkbox",
+      "name": "isCopyIcon",
```
Comment (v-excelsior) [inline components/bl-copyButton-component/component.json:57]: isCopyIconVisible or hasCopyIcon
Same for other

File: components/bl-copyButton-component/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+/dist/
```
Comment (v-excelsior) [inline components/bl-copyButton-component/.gitignore:1]: Could be removed

File: components/bl-copyButton-component/component.json [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  "description": "",
+  "showInToolbox": true,
+  "faIcon": "copy",
+  "mainJS": "dist/index.js",
+  "type": "custom",
+  "category": "Custom Components",
+  "properties": [
+    {
+      "type": "checkbox",
+      "name": "isDisabled",
```
Comment (v-excelsior) [inline components/bl-copyButton-component/component.json:13]: isDisabled -> disabled
isDisabledLogic - disabledLogic

**Review by v-excelsior — CHANGES_REQUESTED** [review]

**Pull request merged**